apps/api/src/server/applications/s51advice/s51-advice.controller.js

```diff
@@ -503,3 +503,31 @@ export const deleteS51Advice = async ({ params: { adviceId } }, response) => {
 
 	response.send(s51Advice);
 };
+
+/**
+ * Unpublish S51 advice item
+ *
+ * @type {import('express').RequestHandler<{adviceId: number}, any, any, any>}
+ */
+export const unpublishS51Advice = async ({ body, params }, response) => {
+	const adviceId = params.adviceId;
+	const payload = body[''];
+
+	const advice = await s51AdviceRepository.get(adviceId);
+	if (!advice) {
+		throw new BackOfficeAppError(`no S51 advice found with id ${adviceId}`, 404);
+	}
+	const docs = await s51AdviceDocumentRepository.getForAdvice(adviceId);
+	advice.S51AdviceDocument = docs;
+	payload.publishedStatusPrev = advice.publishedStatus;
+
+	await eventClient.sendEvents(
+		NSIP_S51_ADVICE,
+		[buildNsipS51AdvicePayload(advice)],
+		EventType.Publish
+	);
```

Comment:
Could be better to run this after you update in the DB, in case the update fails. To make sure the event is describing something that definitely happened.